### ISSUE TITLE
Maxout not implemented

### ISSUE THREAD

**Issue by wmac** [aigamedev/scikit-neuralnetwork#142]
Hello,

1- When I install using pip, the installed version gives the following error when I try to run the classification example on your website (at fitting step):

E:\programming\python\WinPython\python-2.7.10.amd64\lib\site-packages\sknn\backend\pylearn2\__init__.py:13: UserWarning: 

The PyLearn2 backend is deprecated; the next release will switch to Lasagne by default.

Test the change using the following at the top of your script:
> from sknn.backend import lasagne

  """ + ansi.ENDC, category=UserWarning)

2- When I install from your git repository, for the classification, it gives error that "MaxOut" is not implemented. 

This is the classification example that I took from your website:

from sknn.mlp import Classifier, Layer
nn = Classifier(
    layers=[
        Layer("Maxout", units=100, pieces=2),
        Layer("Softmax")],
    learning_rate=0.001,
    n_iter=25)

Thanks.

**Comment by wmac** [comment]
Please disregard the warning part as it has already been discussed.

**Comment by alexjc** [comment]
Maxout is unlikely to return. It has been replaced by other activation functions like leaky RELU, parametric PRELU, and more recently exponential linear ELU.  ELU are supported in the master branch as "ExpLin" activation. I recommend you switch to those, as only PyLearn2 seems to use Maxout and that is no longer maintained.

I'm going to close this issue (won't fix), but feel free to ask anything else if you want.

**Comment by kthouz** [comment]
Hi I am using sknn for the first time and I got through a similar problem. I am training with a dataset that has 12 output classes. I used 'Maxout' as in the example above and i returned NotImplemented Error. When I switched to ExpLin, the model can only detect two classes of the 12. Did you finally get a way around this?

**Comment by wmac** [comment]
You should use Softmax for your classifier output (the sum of outputs will be 1). The example on the website could be updated with:

nn = Classifier(
    layers=[
        Layer("Sigmoid", units=100),
        Layer("Softmax")],
    learning_rate=0.001,
    n_iter=25)

**Comment by alexjc** [comment]
PR is welcome for the docs!
